lib/bootloader_pvm.pm

```diff
@@ -67,6 +67,90 @@ sub get_into_net_boot {
     assert_screen ["pvm-grub", "novalink-failed-first-boot"];
 }
 
+sub boot_pvm {
+    if (check_var('BACKEND', 'spvm')) {
+        boot_spvm();
+    } elsif (check_var('BACKEND', 'pvm_hmc')) {
+        boot_hmc_pvm();
+    }
+}
+
+sub boot_hmc_pvm {
+    my $hmc_machine_name = get_required_var('HMC_MACHINE_NAME');
+    my $lpar_id          = get_required_var('LPAR_ID');
+    my $hmc              = select_console 'powerhmc-ssh';
+
+    # detach possibly attached terminals - might be left over
+    type_string "rmvterm -m $hmc_machine_name --id $lpar_id && echo 'DONE'\n";
+    assert_screen 'pvm-vterm-closed';
+
+    # power off the machine if it's still running - and don't give it a 2nd chance
+    type_string "chsysstate -r lpar -m $hmc_machine_name -o shutdown --immed --id $lpar_id && echo 'LPAR SUCCESSFULLY SHUT DOWN'\n";
+    assert_screen [qw(pvm-poweroff-successful pvm-poweroff-not-running)], 180;
+
+    # proceed with normal boot if is system already installed, use sms boot for installation
+    my $bootmode = get_var('BOOT_HDD_IMAGE') ? "norm" : "sms";
+    type_string "chsysstate -r lpar -m $hmc_machine_name -o on -b ${bootmode} --id $lpar_id && echo 'LPAR SUCCESSFULLY BOOTED'\n";
+    assert_screen "pvm-poweron-successful";
+
+    # don't wait for it, otherwise we miss the menu
+    type_string "mkvterm -m $hmc_machine_name --id $lpar_id\n";
```

Comment:
Just curios about booting to system without running installer, current `boot_spvm` contains `return` at this point before  `get_into_net_boot`. Will it be supported as well?

Reply:
First I thought it makes no sense to have it but after talking to you I've implemented this feature in https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/9016/commits/3487e11c0895e6d98837e1342b7c6c531d5791f2 :)